sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -197,6 +198,33 @@ object GpuParquetScanBase {
         meta.willNotWorkOnGpu(s"$other is not a supported read rebase mode")
     }
   }
+
+  private[rapids] def convertDecimal32Columns(t: Table): Table = {
+    val containDecimal32Column = (0 until t.getNumberOfColumns).exists { i =>
+      t.getColumn(i).getType.getTypeId == DType.DTypeEnum.DECIMAL32
+    }
+    // return input table if there exists no DECIMAL32 columns
+    if (!containDecimal32Column) return t
+
+    val columns = (0 until t.getNumberOfColumns).map { i =>
+      t.getColumn(i).getType match {
+        case tpe if tpe.getTypeId == DType.DTypeEnum.DECIMAL32 =>
+          t.getColumn(i).castTo(DType.create(DType.DTypeEnum.DECIMAL64, tpe.getScale))
+        case _ =>
+          t.getColumn(i)
+      }
+    }
+    val ret = new Table(columns: _*)
+    // clean temporary column vectors produced by castTo
+    (0 until t.getNumberOfColumns).foreach {
+      case i if t.getColumn(i).getType.getTypeId == DType.DTypeEnum.DECIMAL32 =>
+        columns(i).close()
+      case _ =>
+    }
+    // clean original table
+    t.close()
```

Comment:
All of this can cause leaks in the case of an exception. I think we can fix it with very little change though.